### ISSUE TITLE
zfs change-key <fs> causes kernel panic on one (of many) encrypted filesystems

### ISSUE THREAD

**Issue by jstenback** [zfsonlinux/zfs#9808]
### System information

Type | Version/Name
 --- | --- 
Distribution Name	| Arch
Distribution Version	| current, lts kernel
Linux Kernel	| 4.19.92-1-lts
Architecture	| x86_64
ZFS Version	| 0.8.2-1
SPL Version	| 0.8.2-1

### Describe the problem you're observing

zfs change-key <fs> causes kernel panic on one of my encrypted filesystems (I have many, only one exhibits this problem). See journal log info below.

### Describe how to reproduce the problem

Unclear, except in my case on one of my systems this reproduces every time with this one particular filesystem.

The one thing that sets this filesystem apart from the others I have where I don't see this problem on this host is that it contains zfs volumes that are *not* encrypted.

### Include any warning/errors/backtraces from the system logs

```
kernel:  ret_from_fork+0x35/0x40
kernel:  ? kthread_park+0x80/0x80
kernel:  ? __thread_exit+0x20/0x20 [spl]
kernel:  kthread+0xfb/0x130
kernel:  thread_generic_wrapper+0x6f/0x80 [spl]
kernel:  ? txg_thread_exit.isra.0+0x60/0x60 [zfs]
kernel:  ? __switch_to+0xc5/0x490
kernel:  txg_sync_thread+0x2d2/0x4b0 [zfs]
kernel:  ? spa_txg_history_init_io+0x123/0x130 [zfs]
kernel:  spa_sync+0x5a9/0xf90 [zfs]
kernel:  dsl_pool_sync+0x3e1/0x4d0 [zfs]
kernel:  dsl_sync_task_sync+0xa6/0xf0 [zfs]
kernel:  spa_keystore_change_key_sync+0x161/0x3c0 [zfs]
kernel:  spa_keystore_change_key_sync_impl+0x1e5/0x320 [zfs]
kernel:  spa_keystore_change_key_sync_impl+0x1e5/0x320 [zfs]
kernel:  spa_keystore_change_key_sync_impl+0x253/0x320 [zfs]
kernel:  ? dsl_dir_hold_obj+0xad/0x3f0 [zfs]
kernel:  ? dmu_buf_replace_user+0x4d/0x70 [zfs]
kernel:  spl_panic+0xd4/0xfc [spl]
kernel:  dump_stack+0x66/0x90
kernel: Call Trace:
kernel: Hardware name: Supermicro Super Server/X10DRL-i, BIOS 2.0 12/18/2015
kernel: CPU: 23 PID: 3002 Comm: txg_sync Tainted: P         C OE     4.19.92-1-lts #1
kernel: Showing stack for process 3002
kernel: PANIC at dsl_crypt.c:1448:spa_keystore_change_key_sync_impl()
kernel: VERIFY3(0 == dsl_dir_get_encryption_root_ddobj(dd, &curr_rddobj)) failed (0 == 2)
```

**Comment by behlendorf** [comment]
This is a duplicate of #9524 which was resolved by commit bae11ba8dc1dd992643dd0ab0b5cc3b3cc197fe9.  This fix is queued up for the pending 0.8.3 tag.

**Comment by jstenback** [comment]
Ah, indeed! Apologies for not finding that one before filing this issue. Thanks!
